app/models/pipeline_run.rb

```diff
@@ -723,7 +729,7 @@ def generate_aggregate_counts(tax_level_name)
     uncategorizable_id = TaxonLineage::MISSING_LINEAGE_ID.fetch(tax_level_name.to_sym, -9999)
     uncategorizable_name = "Uncategorizable as a #{tax_level_name}"
     TaxonCount.connection.execute(
-      "INSERT INTO taxon_counts(pipeline_run_id, tax_id, name,
+      "REPLACE INTO taxon_counts(pipeline_run_id, tax_id, name,
```

Comment:
Trying to understand better... wouldn't the block you added above remove the need for "replace"?

Reply:
because we have NT/NT+ we kinda need this.

Reply:
Isn't it handled by "count_type"?

Reply:
no. because this function will be called twice and the second time NT/NR records are already there, this will cause an insert error.

Reply:
ok